src/components/kalman_component/kalman.sh

```diff
@@ -0,0 +1,151 @@
+    python ${InversionPath}/src/components/kalman_component/change_dates.py $StartDate_i $EndDate_i $JacobianRunsDir; wait
+    python ${InversionPath}/src/components/kalman_component/change_dates.py $StartDate_i $EndDate_i $PosteriorRunDir; wait
+    echo "Edited Start/End dates in geoschem_config.yml for prior/perturbed/posterior simulations: $StartDate_i to $EndDate_i"
+
+    # Prepare initial (prior) emission scale factors for the current period
+    ConfigPath=${InversionPath}/${ConfigFile}
+    echo "python path = $PYTHONPATH"
+    python ${InversionPath}/src/components/kalman_component/prepare_sf.py $ConfigPath $i ${RunDirs} $NudgeFactor; wait
+
+    ##=======================================================================
+    ##  Submit all Jacobian simulations OR submit only the Prior simulation
+    ##=======================================================================
+    
+    # run jacobian simulation for the given period
+    run_jacobian
+
+    # run inversion for the given period
+    run_inversion
+
+    # Update ScaleFactor.nc with the new posterior scale factors before running the posterior simulation
+    # NOTE: This also creates the posterior_sf_period{i}.nc file in archive_sf/
+    python ${InversionPath}/src/components/kalman_component/multiply_posteriors.py $i ${RunDirs}; wait
+    echo "Multiplied posterior scale factors over record"
+
+    # Print total posterior emissions
+    python ${InversionPath}/src/components/kalman_component/print_posterior_emissions.py $ConfigPath $i ${RunDirs}; wait
+
+    run_posterior
+
+    # Make a copy of the posterior output/diags files for postproc_diags.py
+    copydir="${PosteriorRunDir}/OutputDir"
+    cp ${copydir}/GEOSChem.SpeciesConc.${EndDate_i}_0000z.nc4 ${copydir}/GEOSChem.SpeciesConc.Copy.${EndDate_i}_0000z.nc4
+    cp ${copydir}/GEOSChem.LevelEdgeDiags.${EndDate_i}_0000z.nc4 ${copydir}/GEOSChem.LevelEdgeDiags.Copy.${EndDate_i}_0000z.nc4
+    echo "Made a copy of the final posterior SpeciesConc and LevelEdgeDiags files"
+
+    # Make link to restart file from posterior run directory in each Jacobian run directory
+    for ((x=0;x<=nElements;x++)); do
+        # Add zeros to string name
+        if [ $x -lt 10 ]; then
+            xstr="000${x}"
+        elif [ $x -lt 100 ]; then
+            xstr="00${x}"
+        elif [ $x -lt 1000 ]; then
+            xstr="0${x}"
+        else
+            xstr="${x}"
+        fi
+        ln -sf ${PosteriorRunDir}/Restarts/GEOSChem.Restart.${EndDate_i}_0000z.nc4 ${JacobianRunsDir}/${RunName}_${xstr}/Restarts/.
+    done
+    echo "Copied posterior restart to $((x-1)) Jacobian run directories for next iteration"
+
+    cd ${InversionPath}
+
+    # Delete unneeded daily restart files from Jacobian and posterior directories
+    # python ${InversionPath}/src/components/kalman_component/cull_restarts.py $JacobianRunsDir $PosteriorRunDir $StartDate_i $EndDate_i
```

Comment:
This is where we could call the `cull_restarts()` function. I can't remember why I commented it out -- perhaps just in testing my original branch...

Reply:
I'll put this back in!